src/vmm/src/vstate/vm.rs

```diff
@@ -34,6 +246,12 @@ pub struct VmCommon {
     max_memslots: usize,
     /// The guest memory of this Vm.
     pub guest_memory: GuestMemoryMmap,
+    /// Interrupts used by Vm's devices
+    pub interrupts: Arc<Mutex<HashMap<u32, RoutingEntry>>>,
+    /// Allocator for VM resources
+    pub resource_allocator: Arc<ResourceAllocator>,
+    /// MMIO bus
+    pub mmio_bus: Arc<vm_device::Bus>,
```

Comment:
These fields should not be here. They all should be in the `DeviceManager` and it should be responsible for all interrupt/MSI/relocation handling. Since the `VmFd` will still be needed there, we can just copy it in the device manager (only `VmFd`, not the whole `Vm` struct)

Reply:
After offline discussion, I was convinced that moving all `Vm` related things (like `DeviceManager`, `vcpus` etc) into `Vm` type is a good solution. Current approach, though not perfect, will be updated in later PR.